app/src/main/java/com/dragic/beggining_recyclerview_kodeco/ui/CreatureAdapter.kt

```diff
@@ -1,56 +1,83 @@
 package com.dragic.beggining_recyclerview_kodeco.ui
 
+import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.view.animation.AnimationUtils
 import androidx.recyclerview.widget.DiffUtil
 import androidx.recyclerview.widget.ListAdapter
 import androidx.recyclerview.widget.RecyclerView
 import com.dragic.beggining_recyclerview_kodeco.R
-import com.dragic.beggining_recyclerview_kodeco.app.inflate
 import com.dragic.beggining_recyclerview_kodeco.databinding.ListItemCreatureBinding
+import com.dragic.beggining_recyclerview_kodeco.databinding.ListItemPlanetHeaderBinding
+import com.dragic.beggining_recyclerview_kodeco.model.CompositeItem
 import com.dragic.beggining_recyclerview_kodeco.model.Creature
+import java.lang.IllegalArgumentException
 
 class CreatureItemDiffCallback : DiffUtil.ItemCallback<Creature>() {
     override fun areItemsTheSame(oldItem: Creature, newItem: Creature): Boolean = oldItem == newItem
 
     override fun areContentsTheSame(oldItem: Creature, newItem: Creature): Boolean = oldItem.id == newItem.id
 }
 
-class CreatureAdapter(private val creatures: MutableList<Creature>, val onCreatureSelect: ((Int) -> Unit)) :
-    ListAdapter<Creature, CreatureAdapter.ViewHolder>(CreatureItemDiffCallback()) {
+class CreatureAdapter(private val compositeItems: MutableList<CompositeItem>, val onCreatureSelect: ((Int) -> Unit)) :
```

Comment:
Passing mutable lists around is bad practice because you're no longer in control of where it changes. A scenario like this is possible: 

```kt
val adapterList = mutableListOf("item1", "item2")
val adapter = CreatureAdapter(adapterList, ...)
...
adapterList.remove("item1")
```

The adapter now doesn't know that the has list changed. 

To fix this, you should refactor CreatureAdapter to take a List<Creature> instead, and to keep a private val MutableList<Creature> that's initialized with the immutable list from the constructor.

Reply:
done